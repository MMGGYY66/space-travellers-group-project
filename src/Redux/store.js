import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './Missions/missions';
import rocketReducer from './Rockets/Rockets';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    missions: missionsReducer,
  },
});

export default store;
