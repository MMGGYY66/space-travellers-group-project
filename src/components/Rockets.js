import React from 'react';

const Rocket = () => (
  <h1>Rocket List</h1>
);

export default Rocket;
