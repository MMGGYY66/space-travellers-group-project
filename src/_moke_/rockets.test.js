import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
import Rockets from '../pages/Rockets';
import Rocket from '../components/Rocket';
import RocketTest from './rocketsReducer';

const rocket = {
  rocketName: 'Heloo', description: 'Heloof', flickrImages: 'Helookn', reserved: false, id: 2,
};

jest.mock('axios');
describe('Rockets must pass the test', () => {
  test('added Rockets must return data', () => {
    expect(RocketTest.addedRockets()).toBeDefined();
  });
  test('added Rockets return value length must be 4', () => {
    expect(RocketTest.addedRockets()).toHaveLength(4);
  });
  test('added Rockets must return name t', () => {
    expect(RocketTest.addedRockets()[3].description).toBe('easy taske done today');
  });
});

describe('Rockets component', () => {
  it('renders Rockets component', () => {
    const r = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Rockets />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(r).toMatchSnapshot();
  });
  it('renders Rocket component', () => {
    const m = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Rocket rocket={rocket} />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(m).toMatchSnapshot();
  });
});
