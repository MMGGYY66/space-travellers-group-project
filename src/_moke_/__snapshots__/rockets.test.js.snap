// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component renders Rocket component 1`] = `
<div
  className="d-flex py-3 px-5 height gap-5"
>
  <img
    alt="rockets img"
    className="w-25 h "
    src="Helookn"
  />
  <div
    className="w-75 gap-2"
  >
    <h2>
      Heloo
    </h2>
    <p>
      Heloof
    </p>
    <button
      className="btn btn-primary mb-2"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;

exports[`Rockets component renders Rockets component 1`] = `
<div
  className="mt-3"
/>
`;
